database/postgres/subscription_test.go

```diff
@@ -542,6 +542,75 @@ func Test_FindCLISubscriptions(t *testing.T) {
 	require.Equal(t, 8, len(dbSubs))
 }
 
+func Test_DeliveryModes(t *testing.T) {
+	db, closeFn := getDB(t)
+	defer closeFn()
+
+	subRepo := NewSubscriptionRepo(db)
+	project := seedProject(t, db)
+	source := seedSource(t, db)
+	endpoint := seedEndpoint(t, db)
+
+	testCases := []struct {
+		name                string
+		deliveryMode        datastore.DeliveryMode
+		expectedInitial     datastore.DeliveryMode
+		updateTo            datastore.DeliveryMode
+		expectedAfterUpdate datastore.DeliveryMode
+	}{
+		{
+			name:                "At Least Once",
+			deliveryMode:        datastore.AtLeastOnceDeliveryMode,
+			expectedInitial:     datastore.AtLeastOnceDeliveryMode,
+			updateTo:            datastore.AtMostOnceDeliveryMode,
+			expectedAfterUpdate: datastore.AtMostOnceDeliveryMode,
+		},
+		{
+			name:                "At Most Once",
+			deliveryMode:        datastore.AtMostOnceDeliveryMode,
+			expectedInitial:     datastore.AtMostOnceDeliveryMode,
+			updateTo:            datastore.AtLeastOnceDeliveryMode,
+			expectedAfterUpdate: datastore.AtLeastOnceDeliveryMode,
+		},
+		{
+			name:                "Empty String",
+			deliveryMode:        "",
+			expectedInitial:     datastore.AtLeastOnceDeliveryMode, // default value
+			updateTo:            datastore.AtMostOnceDeliveryMode,
+			expectedAfterUpdate: datastore.AtMostOnceDeliveryMode,
+		},
+	}
```

Comment:
Test nil `expectedInitial` too?
```suggestion
		{
			name:                "Nil",
			deliveryMode:        nil,
			expectedInitial:     datastore.AtLeastOnceDeliveryMode, // default value
			updateTo:            datastore.AtMostOnceDeliveryMode,
			expectedAfterUpdate: datastore.AtMostOnceDeliveryMode,
		},
	}
```